src/subcommands/deployments.ts

```diff
@@ -212,98 +216,20 @@ async function listDeployments(args: Args): Promise<void> {
-    // Default to showing the production deployment of the project
-    if (!projectIdArg) {
-      error(
-        "No deployment or project specified. Use --id <deployment-id> or --project <project-name>",
-      );
-    }
-    projectId = projectIdArg;
-    const spinner = wait(
-      `Searching production deployment of project '${projectId}'...`,
-    ).start();
-    project = await api.getProject(projectId);
-    if (!project) {
-      spinner.fail(
-        `The project '${projectId}' does not exist, or you don't have access to it`,
-      );
-      return Deno.exit(1);
-    }
-    if (!project.productionDeployment) {
-      spinner.fail(
-        `Project '${project.name}' does not have a production deployment. Use --id <deployment-id> to specify the deployment to show`,
-      );
-      return Deno.exit(1);
-    }
-    deploymentId = project.productionDeployment.deploymentId;
-    spinner.succeed(
-      `The production deployment of the project '${project.name}' is '${deploymentId}'`,
-    );
-  }
-
-  if (args.prev.length !== 0 || args.next.length !== 0) {
-    // Search the deployment relative to the specified deployment
-    if (!projectId) {
-      // Fetch the deployment specified with --id, to know of which project to search the relative deployment
-      // If user didn't use --id, they must have used --project, thus we already know the project-id
-      const spinner_ = wait(`Fetching deployment '${deploymentId}'...`)
-        .start();
-      const specifiedDeployment = await api.getDeployment(deploymentId);
-      if (!specifiedDeployment) {
-        spinner_.fail(
-          `The deployment '${deploymentId}' does not exist, or you don't have access to it`,
-        );
-        return Deno.exit(1);
-      }
-      spinner_.succeed(`Deployment '${deploymentId}' found`);
-      projectId = specifiedDeployment.project.id;
-    }
-    let relativePos = 0;
-    for (const prev of args.prev) {
-      relativePos -= parseInt(prev || "1");
-    }
-    for (const next of args.next) {
-      relativePos += parseInt(next || "1");
-    }
-    const relativePosString = relativePos.toLocaleString(navigator.language, {
-      signDisplay: "exceptZero",
-    });
-    const spinner = wait(
-      `Searching the deployment ${relativePosString} relative to '${deploymentId}'...`,
-    ).start();
-    const maybeBuild = await searchRelativeDeployment(
-      api.listAllDeployments(projectId),
-      deploymentId,
-      relativePos,
-    );
-    if (!maybeBuild) {
-      spinner.fail(
-        `The deployment '${deploymentId}' does not have a deployment ${relativePosString} relative to it`,
-      );
-      return Deno.exit(1);
-    }
-    build = maybeBuild;
-    spinner.succeed(
-      `The deployment ${relativePosString} relative to '${deploymentId}' is '${build.deploymentId}'`,
-    );
-  }
+  let [deploymentId, projectId, build, project]: [
+    string,
+    string | undefined,
+    Build | null | undefined,
+    Project | null | undefined,
```

Comment:
```suggestion
    Build | undefined,
    Project | undefined,
```

Reply:
Can't do that, cause later we assign the result of `api.getProject(): Project | null` to this variable